ownCloud/Client/Actions/Actions+Extensions/DeleteAction.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  DeleteAction.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 12/11/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+/*
+* Copyright (C) 2018, ownCloud GmbH.
+*
+* This code is covered by the GNU Public License Version 3.
+*
+* For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+* You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+*
+*/
+
+import ownCloudSDK
+
+class DeleteAction : Action {
+	override class var identifier : OCExtensionIdentifier? { return OCExtensionIdentifier("com.owncloud.action.delete") }
+	override class var category : ActionCategory? { return .destructive }
+	override class var name : String? { return "Delete".localized }
+
```

Comment:
In order to make use of the location identifiers and control where an action should eventually turn up, please add 
```swift
override class var locations : [OCExtensionLocationIdentifier]? { return [.moreItem, .tableRow] }
```
to all `Action` subclasses (of course adjusting the returned locations array as needed).